src/ddsrt/include/dds/ddsrt/process.h

```diff
@@ -41,6 +44,98 @@ typedef pid_t ddsrt_pid_t;
 DDS_EXPORT ddsrt_pid_t
 ddsrt_getpid(void);
 
+#ifdef PROCESS_MANAGEMENT_ENABLED
+
+/**
+ * @brief Create new process.
+ *
+ * Creates a new process using the provided executable file. It will have
+ * default priority and scheduling.
+ *
+ * Process arguments are represented by argv, which can be null. If argv is
+ * not null, then the array must be null terminated. The argv array only has
+ * to contain the arguments, the executable filename doesn't have to be in
+ * the first element, which is normally the convention.
+ *
+ * @param[in]   executable     Executable file name.
+ * @param[in]   argv           Arguments array.
+ * @param[out]  pid            ID of the created process.
+ *
+ * @returns A dds_retcode_t indicating success or failure.
+ *
+ * @retval DDS_RETCODE_OK
+ *             Process successfully created.
+ * @retval DDS_RETCODE_BAD_PARAMETER
+ *             Provided file is not available or not executable.
+ * @retval DDS_RETCODE_NOT_ALLOWED
+ *             Caller is not permitted to start the process.
+ * @retval DDS_RETCODE_OUT_OF_RESOURCES
+ *             Not enough resources to start the process.
+ * @retval DDS_RETCODE_ERROR
+ *             Process could not be created.
+ */
+DDS_EXPORT dds_retcode_t
+ddsrt_process_create(
+  const char *executable,
+  char *const argv[],
+  ddsrt_pid_t *pid);
+
+/**
+ * @brief Get the exit code of a process.
+ *
+ * @param[in]   pid            Process ID (PID) to get the exit code from.
+ * @param[out]  code           The exit code of the process.
+ *
+ * @returns A dds_retcode_t indicating success or failure.
+ *
+ * @retval DDS_RETCODE_OK
+ *             Process has terminated and its exit code has been captured.
+ * @retval DDS_RETCODE_PRECONDITION_NOT_MET
+ *             Process is still alive.
+ * @retval DDS_RETCODE_BAD_PARAMETER
+ *             Process unknown.
+ * @retval DDS_RETCODE_ERROR
+ *             Getting the exit code failed for an unknown reason.
+ */
+DDS_EXPORT dds_retcode_t
+ddsrt_process_get_exit_code(
```

Comment:
I'd prefer a separate *ddsrt_proc_exists* and *ddsrt_proc_wait* because they do different things.